src/internal/testutil/identity.go

```diff
@@ -109,6 +109,10 @@ func DoOAuthExchange(t testing.TB, pachClient, enterpriseClient *client.APIClien
 	resp, err := c.Get(RewriteURL(t, loginURL, DexHost(enterpriseClient)))
 	require.NoError(t, err)
 
+	// Dex login redirects to the provider page, which will generate it's own state
```

Comment:
Dex made a change where the incoming state is stored in the DB and a new nonce is created for the login method, which adds a redirect to the login flow. We just need to follow the extra redirect and everything works.